### PR TITLE
Fix register

### PR DESCRIPTION
**Pull request by mikecasas**
This just hides the UI if the site does not allow registration.  I will get the controller later. You can wait until I add that, or accept this as-is.

### CODE REVIEW

**Comment by mikecasas** [comment]
I added the check on the controller.

**Comment by sbwalker** [comment]
@mikecasas rather than using CSS properties which simply make content invisible, it may be better to simply wrap the markup in a @if (PageState.Site.AllowRegistration) { registration markup } else { < div>Registration is Disabled< /div> }. The benefit is that the markup will not even be rendered by the component.

**Comment by sbwalker** [comment]
@mikecasas I can make the change I suggested after I merge this PR if that is your preference?

**Comment by mikecasas** [comment]
Sure your preference, you are the boss :)

File: Oqtane.Client/Services/UserService.cs [hunk 1]
```diff
@@ -36,6 +38,13 @@ public async Task<User> GetUserAsync(string username, int siteId)
 
         public async Task<User> AddUserAsync(User user)
         {
+            Site site = await _siteService.GetSiteAsync(_siteState.Alias.SiteId, _siteState.Alias);
```
Comment (chlupac) [inline Oqtane.Client/Services/UserService.cs:41]: Here was bug, repaired.

**Comment by sbwalker** [comment]
I just fixed it in #381

**Pull request merged**